ledger/voters.go

```diff
@@ -312,9 +312,23 @@ func (vt *votersTracker) registerPrepareCommitListener(commitListener ledgercore
 	vt.commitListenerMu.Lock()
 	defer vt.commitListenerMu.Unlock()
 
+	if vt.commitListener != nil {
+		vt.l.trackerLog().Error("votersTracker.registerPrepareCommitListener: overriding existing listener! commitListener should have been nil'd on stateproof.Worker.Shutdown().")
+	}
 	vt.commitListener = commitListener
 }
 
+func (vt *votersTracker) unregisterPrepareCommitListener() {
+	vt.commitListenerMu.Lock()
+	defer vt.commitListenerMu.Unlock()
+
+	if vt.commitListener == nil {
+		vt.l.trackerLog().Warn("votersTracker.unregisterPrepareCommitListener: no listener was registered.")
```

Comment:
this might add lots of warning messages to logs since tests do not register this listener